src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -102,47 +107,35 @@ public sealed class PeerConnector : IPeerConnector
         public RelatedPeerConnectors RelatedPeerConnector { get; set; }
 
         /// <inheritdoc/>
-        public NetworkPeerRequirement Requirements { get; private set; }
+        public NetworkPeerRequirement Requirements { get; internal set; }
 
         /// <summary>Constructor used for unit testing.</summary>
-        internal PeerConnector(
-            IPeerAddressManager peerAddressManager,
-            PeerIntroductionType peerIntroductionType)
+        protected PeerConnector(NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
         {
+            Guard.NotNull(nodeSettings, nameof(nodeSettings));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
 
-            this.nodeLifetime = new NodeLifetime();
             this.peerAddressManager = peerAddressManager;
-            this.peerIntroductionType = peerIntroductionType;
+            this.NodeSettings = nodeSettings;
+            this.nodeLifetime = new NodeLifetime();
```

Comment:
Ouch mate... 😮

Reply:
Yes this constructor is only used for unit testing ;)

Reply:
If you create a test factory in a proper way you wont need this constructor for testing